amf-webapi/shared/src/main/scala/amf/plugins/document/webapi/resolution/stages/ExtensionDomainElementMerge.scala

```diff
@@ -104,7 +104,7 @@ class ExtensionDomainElementMerge(restrictions: MergingRestrictions,
       case amfObject: AmfObject => (amfObject.id, amfObject.annotations)
       case array: AmfArray =>
         val ann =
-          if (value.annotations.nonEmpty) value.annotations
+          if (value.annotations.nonEmpty && !value.annotations.contains(classOf[VirtualNode])) value.annotations
```

Comment:
Extract to function?

Reply:
The virtual Node is an interface for any kind of virtual annotations, so I'm not sure if is good idea to generalize it.
Good options could be that the field is inferred or an AmfElement is a virtual element, but the general annotation as virtual sounds confused.